src/python/pants/rules/core/core_test_model.py

```diff
@@ -18,6 +19,7 @@ class TestResult:
   status: Status
   stdout: str
   stderr: str
+  coverage_digest: Digest
```

Comment:
This should probably be `Optional[Digest]`.

Also, do you know how other languages handle coverage? Do they also generate files? I'm wondering if this change would be generalizable to something like Junit. A quick scan of how Junit, Mocha, and/or Scalatest do test coverage would be helpful.

Reply:
Coverage is depressingly language-specific :( But it looks like the world is gradually standardising on Cobertura XML (in the same way that JUnit XML gradually took over test results)... If we're aiming to have one standard format across languages, it may be worth naming this explicitly `cobertura_coverage_digest` (and having it be `Optional`)

Reply:
In the case of python coverage, this digest is of a sqllite data file.

Reply:
I _think_ if you pass `--cov-report=xml` it will generate Cobertura-compatible XML, which could interop with other languages.

Reply:
Yes, but it will only do that for the test target it's running over, and I cannot merge together the report files. `coverage merge` requires the sqllite data. 

I'll have a couple more commits up later today that will make this more clear - the next two operations are merging together the coverage data from each test target and generating a report, which can absolutely be the Cobertura-compatible XML.